File: test_frontend_mapping.js
const fs = require('fs');

async function testParsing() {
  const json = await fetch('http://localhost:8080/api/zerion-proxy?address=0x2ba20848f5125dd14b2620d978f83a993176a43f&filter=all').then(r=>r.json());
  const data = json.data || [];
  console.log(`Initial items from Proxy: ${data.length}`);

  let aeroCount = 0;
  const positions = [];

  for (const item of data) {
    const attr = item.attributes;
    if (!attr) continue;
    
    // simulate portfolioService logic
    let protocolName = attr.application_metadata?.name ?? attr.protocol ?? (attr.position_type === 'wallet' ? 'Wallet' : 'Unknown');

    if (
      (attr.name && attr.name.toLowerCase().includes('aerodrome')) || 
      (item.id && item.id.toLowerCase().includes('aerodrome'))
    ) {
      protocolName = 'Aerodrome Slipstream';
      aeroCount++;
    }

    const valueUsd = attr.value ?? 0;
    
    // Simulate dust filter
    if (valueUsd <= 0) {
      if (protocolName === 'Aerodrome Slipstream') {
        console.log(`Filtered out aero position because valueUsd is ${valueUsd}`);
      }
      continue;
    }

    positions.push({
      id: item.id,
      name: attr.name,
      protocol: protocolName,
      value: valueUsd
    });
  }

  console.log(`Total aero detected: ${aeroCount}`);
  console.log(`Positions surviving filters: ${positions.length}`);
  const aeroSurviving = positions.filter(p => p.protocol.includes('Aero')).length;
  console.log(`Aero positions surviving: ${aeroSurviving}`);
  
  if (aeroSurviving > 0) {
    console.log('Surviving Aerodrome LPs:', positions.filter(p => p.protocol.includes('Aero')));
  }
}

testParsing().catch(console.error);

File: test_frontend_mapping.py
import json

with open('zerion_complex.json', 'r') as f:
    data = json.load(f)

items = data.get('data', [])
print(f"Loaded {len(items)} items")

positions = []
for item in items:
    attr = item.get('attributes', {})
    
    # 1. Simulate dust filter
    value_usd = attr.get('value')
    if value_usd is None:
        value_usd = 0
        
    # Wait, the frontend says: if (valueUsd <= 0) continue;
    if value_usd <= 0:
        continue
        
    # 2. Simulate protocol mapping
    protocol_name = "Unknown"
    app_meta = attr.get('application_metadata')
    if app_meta and app_meta.get('name'):
        protocol_name = app_meta['name']
    elif attr.get('protocol'):
        protocol_name = attr['protocol']
    elif attr.get('position_type') == 'wallet':
        protocol_name = 'Wallet'
        
    name = attr.get('name', '')
    item_id = item.get('id', '')
    
    if (name and 'aerodrome' in name.lower()) or (item_id and 'aerodrome' in item_id.lower()):
        protocol_name = 'Aerodrome Slipstream'
        
    positions.push_or_append = True
    positions.append({
        'id': item_id,
        'protocol': protocol_name,
        'valueUsd': value_usd
    })

print(f"Positions surviving dust filter: {len(positions)}")
for p in positions:
    print(p)
